exit_test.go

```diff
@@ -0,0 +1,47 @@
+// Copyright 2020 The Kubernetes Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package klog_test
+
+import (
+	"flag"
+	"fmt"
+	"os"
+
+	"k8s.io/klog/v2"
+)
+
+func ExampleFlushAndExit() {
```

Comment:
Any reason why we can't have a TestFlushAndExit (Similar to functions in TestErrorS in klog_test.go) ?

We can check against the following format. 

`format: "\"exiting...\"\nos.Exit(10)"`

Reply:
We could have such a test function, but an example is more useful:
- it gets tested by `go test`
- it gets included in the documentation